### DISCUSSION TITLE
Get delta states on Trace/TraceList, or in Extract

### DISCUSSION THREAD

**Discussion by e-cal** [AI-Planning/macq#48] [General]
For now I will put the function to get delta states in `Extract`, but it could also make sense to put the method on `Trace` and a wrapper function on `TraceList`. 

Also would we prefer the function to take a single action and get all the delta states for the one action, or to always get all delta states for all actions in a `Trace` / `TraceList`?

**Comment by haz** [comment]
An interesting question...for now, let's put it on the State object, and if that's in an observation (like the identity observation), then you'll be able to use it in extract. So it'd be `state.diff_from(other_state)` (I think just using `diff` would be ambiguous).

**Reply by e-cal** [reply]
Sounds good!
